src/test/java/com/sst/utopia/booking/model/TicketTest.java

```diff
@@ -0,0 +1,65 @@
+package com.sst.utopia.booking.model;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case to ensure that Ticket invariant-validation code works properly.
+ *
+ * @author Jonathan Lovelace
+ */
+public class TicketTest {
+
+	@Test
+	public final void testClearingReserverClearsPrice() {
+		final Ticket ticket = new Ticket();
+		ticket.setReserver(new User());
+		ticket.setPrice(500);
+		ticket.setReserver(null);
+		assertNull(ticket.getPrice(), "Setting reserver to null sets price to null");
+	}
+
+	@Test
+	public final void testClearingReserverClearsTimeout() {
+		final Ticket ticket = new Ticket();
+		ticket.setReserver(new User());
+		ticket.setReservationTimeout(LocalDateTime.now());
+		ticket.setReserver(null);
+		assertNull(ticket.getReservationTimeout(),
+				"Setting reserver to null sets reservation timeout to null");
+	}
+
+	@Test
+	public final void testNoPriceWithoutReservation() {
+		assertThrows(IllegalStateException.class, () -> new Ticket().setPrice(500),
+				"Can't set price-paid for an unbooked ticket");
```

Comment:
I am not sure how this succeed with the logic in the setPrice method

Reply:
Same with setting the timeout

Reply:
The `setPrice` and `setReservationTimeout` methods throw exceptions if they are called with non-null arguments when `reserver` is null; that's what these test methods are testing.

Reply:
Yes, you are right.